build.gradle

```diff
@@ -317,19 +317,23 @@ project(':iceberg-flink') {
     compile project(':iceberg-parquet')
     compile project(':iceberg-hive-metastore')
 
+    compileOnly "org.apache.flink:flink-connector-base"
```

Comment:
Why compileOnly? Does this assume that flink-connector-base would be supplied somehow? If so, what's recommended to users of the library given that flink-dist doesn't bundle flink-connector-base.

Reply:
All the Flink deps are defined as `compileOnly` in Iceberg. Yeah, it assumes Flink jars are provided at runtime.

Reply:
This particular jar file won't be provided by the Flink dist. It should be a transitive dependency of the connector.

Reply:
Right now, the `iceberg-flink-runtime` shadow jar doesn't bring in any Flink deps. if we include `flink-connector-base` as `compile`, then it will be bundled in the `iceberg-flink-runtime` shadow. if a Flink app pulls in  `flink-connector-base` transitively via other deps (like Flink Kafka connector), then we can get dup classes in jars.

@openinx maybe you can share some lights on how users get the Flink jars when using the Flink Iceberg connector.

Reply:
I am also wondering if flink-dist should actually include `flink-connector-base`

Reply:
https://issues.apache.org/jira/browse/FLINK-20098

It is not desirable to place such dependencies into flink-dist.

Regarding the transitive dependency: it would be surprising for the user to find that they have to add a `flink-connector-base` dependency to their project for the iceberg connector to work.

Reply:
Regarding the dup classes: The the user still has control over the transitive dependency if there is a version mismatch.

Reply:
@tweise thx a lot for the context. now it all makes sense to me. Also didn't notice that iceberg-flink-runtime actually exclude all flink jars. Updated with compile deps.

In the future, if Flink decided to move `flink-connector-base` and `flink-connector-files` into `flink-dist` (as hinted in FLINK-20472), we can revisit the compile dep status.